src/Stack/stack.cs

```diff
@@ -2,50 +2,80 @@ namespace StackTask
 {
     public class Stack<T>
     {
-        private class StackEntry<TEntry>{
-            public StackEntry<TEntry>? next=null;
+        private class StackEntry<TEntry>
+        {
+            public StackEntry<TEntry>? next = null;
             public TEntry? data;
         }
 
-        private StackEntry<T>? top=null;
+        private StackEntry<T>? top = null;
 
-        public void Push(T value){
-            StackEntry<T> entry=new StackEntry<T>{data=value};
-            if(top==null){
-                top=entry;
-            }
-            else {
-                entry.next=top;
-                top=entry;
+        public void Push(T value)
+        {
+            if (value != null)
```

Comment:
I think we are deciding for the user of the stack. If I'm a user of the stack I assume a value will be pushed no matter what is the value is. Also I would expect the size of the stack to increase.

Reply:
we could discuss different ways to handle null values